.github/workflows/openms_ci_matrix_full.yml

```diff
@@ -0,0 +1,565 @@
+# This is a basic workflow to help you get started with Actions
```

Comment:
document a bit what it does.
e.g., what happens if someone opens a PR against develop?
does this also build packages? where is stuff uploaded to? etc.